src/main/java/com/synopsys/integration/detect/lifecycle/run/step/ContainerScanStepRunner.java

```diff
@@ -0,0 +1,130 @@
+        scanId = operationRunner.uploadBdioHeaderToInitiateScan(blackDuckRunData, bdioHeaderFile);
+        if (scanId == null) {
+            logger.warn("Scan ID was not found in the response from the server.");
+            throw new IntegrationException("Scan ID was not found in the response from the server.");
+        }
+        String scanIdString = scanId.toString();
+        logger.debug("Scan initiated with scan service. Scan ID received: {}", scanIdString);
+    }
+
+    public void uploadImageToStorageService() throws IntegrationException {
+        String storageServiceEndpoint = String.join("", STORAGE_CONTAINERS_ENDPOINT, scanId.toString());
+        logger.debug("Uploading container image artifact to storage endpoint: {}", storageServiceEndpoint);
+
+        try (Response response = operationRunner.uploadFileToStorageService(
+            blackDuckRunData,
+            storageServiceEndpoint,
+            containerImage,
+            STORAGE_IMAGE_CONTENT_TYPE
+        )
+        ) {
+            if (response.isStatusCodeSuccess()) {
+                logger.debug("Container scan image uploaded to storage service.");
+            } else {
+                logger.trace("Unable to upload container image. {} {}", response.getStatusCode(), response.getStatusMessage());
+                throw new IntegrationException(String.join(" ", "Unable to upload container image. Response code:", String.valueOf(response.getStatusCode()), response.getStatusMessage()));
+            }
+        } catch (IOException | IntegrationException e) {
+            throw new IntegrationException(e);
+        }
+    }
+
+    public void uploadImageMetadataToStorageService() throws IntegrationException {
+        String storageServiceEndpoint = String.join("", STORAGE_CONTAINERS_ENDPOINT, scanId.toString(), "/message");
+        logger.debug("Uploading container image metadata to storage endpoint: {}", storageServiceEndpoint);
+
+        JsonObject imageMetadataObject = operationRunner.createContainerScanImageMetadata(scanId, projectNameVersion);
+
+        try (Response response = operationRunner.uploadJsonToStorageService(
+            blackDuckRunData,
+            storageServiceEndpoint,
+            imageMetadataObject.toString(),
+            STORAGE_IMAGE_METADATA_CONTENT_TYPE
+        )
+        ) {
+            if (response.isStatusCodeSuccess()) {
+                logger.debug("Container scan image metadata uploaded to storage service.");
+            } else {
+                logger.trace("Unable to upload container image metadata." + response.getStatusCode() + " " + response.getStatusMessage());
+                throw new IntegrationException(String.join(" ", "Unable to upload container image metadata. Response code:", String.valueOf(response.getStatusCode()), response.getStatusMessage()));
+
+            }
+        } catch (IOException | IntegrationException e) {
```

Comment:
It may be cleaner to allow IntegrationException bubble up instead of trying to catch it, wrap it, and re-throw.

Reply:
Cleaned that up and added to method signature as the catch block wasn't adding anything new here.